### ISSUE TITLE
String out of index error for new aadhaar QR data

### ISSUE THREAD

**Issue by samruddhisk** [tanmoysrt/pyaadhaar#19]
**Describe the bug**
When I try to retrieve the QR data from new aadhaar QR code it throws error as 'string out of index'. Whereas if i try with old aadhaar QR code data, I am getting the correct expected output.

**Expected behavior**
For the new QR Code also i must get the proper result without error. I have updated  my pyaadhar package to 2.0.2 and also my python version is 3.12

**Additional context**
But as per your new release you have mentioned as, it supports the old and new both aadhaar QR codes.
Could you please help me in sorting out this.

**Comment by eshwar-gatram** [comment]
Even I am getting the same error.
@samruddhisk, did you find any solution ????

**Comment by samruddhisk** [comment]
> Even I am getting the same error. @samruddhisk, did you find any solution ????

Yes. I have used the similar logic. But, without the support of pyaadhar package/library. With this approach I am able to fetch the photo as well as other details from the QR code for both old and new versions. I'm sharing my personal repo you can refer that. 
https://github.com/SamuKutre/AADHARQR

**Comment by CODERdotEXE** [comment]
@samruddhisk could you please provide the reference to the actual repository or resources where you found information about the approach?

**Comment by tanmoysrt** [comment]
Hi @samruddhisk 
Can you check please, is the error consists anything related to `six` module or in python <=3.11. Many things deprecated on 3.12.

**Comment by samruddhisk** [comment]
> @samruddhisk could you please provide the reference to the actual repository or resources where you found information about the approach?

Hi @CODERdotEXE, the repository/resource which I have referred to is the same pyaadhaar package by @tanmoysrt. This is the link for the same: https://github.com/tanmoysrt/pyaadhaar/
The logic remains almost similar but, without accessing the library or pyaadhaar package support.

**Comment by samruddhisk** [comment]
> Hi @samruddhisk Can you check please, is the error consists anything related to `six` module or in python <=3.11. Many things deprecated on 3.12.

Okay, I will check and get back to you.

**Comment by CoiferousYogi** [comment]
Hello. I know it is quite late @eshwar-gatram @samruddhisk @tanmoysrt. I can be totally incorrect but it seems the 'string out of index' error stems from Python's own security features. Newer versions of Python from version 3.11 onwards block the default conversion of strings longer than 4300 characters. Please check the length of the string. But it is recommended that you first typecast the string into an integer and then decode it. For security vulnerabilities, you can set a hard limit to the length of the characters beforehand (say 5000) to prevent algorithmic complexity attacks.
